processor/blockprocessor/block_processor_test.go

```diff
@@ -0,0 +1,127 @@
+package blockprocessor_test
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/rpcs"
+	block_processor "github.com/algorand/indexer/processor/blockprocessor"
+	"github.com/algorand/indexer/util/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestProcess(t *testing.T) {
+	//initialize local ledger
+	genesis := test.MakeGenesis()
+	genesisBlock := test.MakeGenesisBlock()
+	initState, err := test.CreateInitState(&genesis, &genesisBlock)
+	if err != nil {
+		log.Panicf("test init err: %v", err)
+	}
+	logger := logging.NewLogger()
+	l, err := ledger.OpenLedger(logger, "local_ledger", true, initState, config.GetDefaultLocal())
```

Comment:
I would create `func makeTestLedger(t *testing.T) *ledger.Ledger` and use it in tests: `pr := block_processor.MakeProcessor(makeTestLedger(t), handler)`. This way boring initialization can be reused across tests.